src/csharp/Microsoft.Spark/Broadcast.cs

```diff
@@ -0,0 +1,243 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization.Formatters.Binary;
+using Microsoft.Spark.Interop;
+using Microsoft.Spark.Interop.Ipc;
+
+namespace Microsoft.Spark
+{
+    /// <summary>
+    /// A broadcast variable. Broadcast variables allow the programmer to keep a read-only variable
+    /// cached on each machine rather than shipping a copy of it with tasks. They can be used, for
+    /// example, to give every node a copy of a large input dataset in an efficient manner. Spark
+    /// also attempts to distribute broadcast variables using efficient broadcast algorithms to
+    /// reduce communication cost.
+    /// </summary>
+    [Serializable]
+    public sealed class Broadcast<T> : IJvmObjectReferenceProvider
+    {
+        [NonSerialized]
+        private readonly JvmObjectReference _jvmObject;
+        private readonly long _bid;
+        [NonSerialized]
+        private readonly string _path;
+
+        internal Broadcast(SparkContext sc,
+            T value,
+            string localDir,
+            JvmObjectReference sparkContext)
```

Comment:
You are arleady passing in `SparkContext`, you don't need to pass in the JvmObjectReference as well.